api/src/api/userRouter.ts

```diff
@@ -136,29 +139,62 @@ export const userRouterFactory = (
       return;
     }
 
-    userDataClient
-      .get(req.params.userId)
-      .then(async (userData: UserData) => {
-        let updatedUserData = userData;
-        updatedUserData = await updateBusinessNameSearchIfNeeded(updatedUserData)
-          .then((userData) => updateOperatingPhase(userData))
-          .then((userData) => updateRoadmapSidebarCards(userData))
-          .then((userData) => asyncUpdateLicenseStatus(userData));
-
-        await userDataClient.put(updatedUserData);
-        res.json(updatedUserData);
-      })
-      .catch((error: Error) => {
-        if (error.message === "Not found") {
-          if (process.env.IS_OFFLINE || process.env.STAGE === "dev") {
-            saveEmptyUserData(req, res, signedInUserId);
+    /* NOTES:
+    We have a number of users who are attempting to log into the application who have not previously visited and
+    therefore do not have any user data. This is causing an infinite loop in the frontend where the user is getting
+    bounced between the onboarding and loading pages. It appears that a tell-tale indicator of such a user is a userId
+    that is an email address.
+
+    TODO: Possible solution - check if the userID is a uuid - if not, log the user out, direct them to onboarding step 1
+    It is likely that this user will enter into a behavior loop where they will then attempt to login again. This is
+    significantly better than the loop behavior that they would otherwise experience. Hopefully, these people will reach
+    out via a support request, at which point we can coach them through the onboarding process.
+    */
+
+    // Check if userId is an email address
+    if (signedInUserId.includes("@")) {
```

Comment:
We had initially talked about responding with a 401 here. Opted not to do that, just capture the log. Thinking was to have a more broad solution and account for a user who does not have any data in our system. This is not something I know of having come up in Intercom previously.

Reply:
(That does cover the case where, like with my user where I deleted my information in DynamoDB, if a user somehow doesn't have information in DDB or it's malformed. But not a top priority to fix right now.)